apps/pie-storybook/stories/pie-button.stories.ts

```diff
@@ -120,91 +120,59 @@ const Template = ({
         </pie-button>
         `;
 
-export const Primary: Story<ButtonProps> = (args: ButtonProps) => Template(args);
-Primary.args = {
-    ...defaultArgs,
-};
+interface ButtonStoryOptions extends StoryOptions {
+    variant?: Variant;
+}
 
-export const Secondary: Story<ButtonProps> = (args: ButtonProps) => Template(args);
-Secondary.args = {
-    ...defaultArgs,
+const createStory = <T>(opts?: ButtonStoryOptions) => ({
```

Comment:
I will abstract this out so it can be used in any story – will do this when I get some more time (and didn't want this PR to get too big)

Reply:
if that's longer than today, please create a ticket or it'll never happen 😂